database/models.py

```diff
@@ -236,3 +233,29 @@ class Document(VersionedBase):
     decision: Mapped[bool]
     decision_date: Mapped[Optional[timestamp]]
     # file
+
+
+class LifeCycleDate(VersionedBase):
+    """
+    Elinkaaritilan päivämäärät
+    """
+
+    __tablename__ = "lifecycle_date"
+
+    lifecycle_status_id: Mapped[uuid.UUID] = mapped_column(
+        ForeignKey("codes.lifecycle_status.id", name="plan_lifecycle_status_id_fkey"),
+        index=True,
+    )
+    plan_id: Mapped[uuid.UUID] = mapped_column(
+        ForeignKey("hame.plan.id", name="plan_id_fkey")
+    )
+    plan_regulation_id: Mapped[uuid.UUID] = mapped_column(
+        ForeignKey("hame.plan_regulation.id", name="plan_regulation_id_fkey")
+    )
+    plan_proposition_id: Mapped[uuid.UUID] = mapped_column(
+        ForeignKey("hame.plan_proposition.id", name="plan_proposition_id_fkey")
+    )
+
```

Comment:
Let's also add nullable relationships here (https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#nullable-many-to-one).

Guess this means something like `plan: Mapped[Optional["Plan"]] = relationship(backref="lifecycle_dates")` etc.